### PR TITLE
zeroex - add celo txns [FOR REVIEW]

### PR DESCRIPTION
**Pull request by RantumBits**
zereoex celo - non-deduped & deduped

### CODE REVIEW

**Comment by RantumBits** [comment]
What's the best way to handle this error? 
Access Denied: Cannot select from table hive.tokens.erc20

File: seeds/zeroex/celo/_schema.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+version: 2
+
+seeds:
+  - name: zeroex_celo_api_fills_sample
+    config:
+      tags: dunesql
+      column_types:
+        # TODO: Add column types here
```
Comment (Hosuke) [inline seeds/zeroex/celo/_schema.yml:8]: Hi @RantumBits,
please define the column types here.
You may move this schema file into `seeds/zeroex/` if you want.

**Comment by RantumBits** [comment]
@Hosuke Can you advise on how to handle error with the dummy legacy model?

File: models/zeroex/celo/zeroex_celo_schema.yml [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+      tags: ["celo","0x","dex_aggregator","dex","aggregator"]
+    description: >
+        0x API erc20 swaps raw fills (deduped) on celo
+    tests:
+      - dbt_utils.unique_combination_of_columns:
+          combination_of_columns:
+            - block_date
+            - tx_hash
+            - evt_index
+      - check_seed_legacy:
```
Comment (Hosuke) [inline models/zeroex/celo/zeroex_celo_schema.yml:100]:
```suggestion
      - check_seed:
```

File: models/zeroex/celo/zeroex_celo_api_fills_deduped.sql [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+      , b.maker_symbol AS maker_symbol
+      , CASE
+        WHEN LOWER(b.taker_symbol) > LOWER(b.maker_symbol)
+        THEN CONCAT(COALESCE(b.maker_symbol, ''), '-', COALESCE(b.taker_symbol, ''))
+        ELSE CONCAT(COALESCE(b.taker_symbol, ''), '-', COALESCE(b.maker_symbol, ''))
+        END AS token_pair   
+      , b.taker_token_amount
+      , b.maker_token_amount
+      , CAST(b.taker_token_amount_raw AS DECIMAL(38,0)) AS taker_token_amount_raw
+      , CAST(b.maker_token_amount_raw AS DECIMAL(38,0)) AS maker_token_amount_raw
```
Comment (Hosuke) [inline models/zeroex/celo/zeroex_celo_api_fills_deduped.sql:84]: Maybe without casting to decimal(38,0) is ok.
```suggestion
      , b.taker_token_amount_raw AS taker_token_amount_raw
      , b.maker_token_amount_raw AS maker_token_amount_raw
```

File: models/zeroex/celo/zeroex_celo_api_fills_deduped.sql [hunk 4]
```diff
@@ -0,0 +1,102 @@
+{{  config(
+        tags=['dunesql'],
+        alias = alias('api_fills_celo_deduped'),
+        materialized='incremental',
+        partition_by = ['block_date'],
```
Comment (Hosuke) [inline models/zeroex/celo/zeroex_celo_api_fills_deduped.sql:5]: We can add a new field `block_month` to use in partition:
```suggestion
        partition_by = ['block_month'],
```

File: models/zeroex/celo/zeroex_celo_api_fills_deduped.sql [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+        , MAX(CASE WHEN maker_consider_flag = 0 THEN NULL ELSE maker_token_amount_raw END)  AS maker_token_amount_raw
+        , COUNT(*)                                                                          AS fills_within
+    FROM fills_first_last a
+    GROUP BY  tx_hash,hop_count
+)
+SELECT  a.blockchain
+      , '0x API'  as project
+      , cast(null as varchar(10)) as version
+      , a.block_date
+      , a.block_time
```
Comment (Hosuke) [inline models/zeroex/celo/zeroex_celo_api_fills_deduped.sql:73]:
```suggestion
      , a.block_time
      , CAST(date_trunc('month', a.block_time) as DATE) as block_month, 
```

File: models/zeroex/celo/zeroex_celo_api_fills.sql [hunk 6]
```diff
@@ -0,0 +1,264 @@
+{{  config(
+    tags=['dunesql'],
+        alias = alias('api_fills_celo'),
+        materialized='incremental',
+        partition_by = ['block_date'],
```
Comment (Hosuke) [inline models/zeroex/celo/zeroex_celo_api_fills.sql:5]: We can use `block_month` here as well.
```suggestion
        partition_by = ['block_month'],
```

**Review by Hosuke — CHANGES_REQUESTED** [review]
Minor suggestions, we can use `block_month` to improve storage efficiency.

**Review by RantumBits — COMMENTED** [review]
ty, all set

**Comment by RantumBits** [comment]
thanks @Hosuke  - all set with these updates

File: models/zeroex/celo/zeroex_celo_schema.yml [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+      - &contract_address
+        name: contract_address
+        desctiption: "The address of the contract which fired the fill/swap event"
+      - &block_time
+        name: block_time
+        description: "Timestamp for block event time in UTC"
+      - &block_date
+        name: block_date
+        description: "UTC event block date of each trade"
+      - &maker
```
Comment (Hosuke) [inline models/zeroex/celo/zeroex_celo_schema.yml:36]:
```suggestion
      - &block_date
        name: block_date
        description: "UTC event block date of each trade"
      - &block_month
        name: block_month
        description: "UTC event block month of each trade"
      - &maker
```

File: models/zeroex/celo/zeroex_celo_schema.yml [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+      - dbt_utils.unique_combination_of_columns:
+          combination_of_columns:
+            - block_date
+            - tx_hash
+            - evt_index
+    columns:
+      - *blockchain
+      - *block_date
+      - *block_time
+      - &taker_symbol
```
Comment (Hosuke) [inline models/zeroex/celo/zeroex_celo_schema.yml:107]:
```suggestion
      - *block_time
      - *block_month
      - &taker_symbol
```

File: seeds/zeroex/celo/zeroex_celo_api_fills_deduped_sample.csv
Comment (Hosuke) [inline seeds/zeroex/celo/zeroex_celo_api_fills_deduped_sample.csv:1]: Seems this seed is currently unused.
You can add a simple `check_seed` test in the schema to use this seed file.

**Review by Hosuke — COMMENTED** [review]
Hi @RantumBits,
it seems these two views do not have any records.
Is this expected?

[Sample zeroex_celo_api_fills](https://dune.com/queries/2989206)
[Sample zeroex_celo_api_fills_deduped](https://dune.com/queries/2989202)

File: seeds/zeroex/celo/zeroex_celo_api_fills_deduped_sample.csv
Comment (RantumBits) [reply seeds/zeroex/celo/zeroex_celo_api_fills_deduped_sample.csv:1]: No, that's not expected. Expecting results like test model: https://dune.com/queries/2755822

Comment (Hosuke) [reply seeds/zeroex/celo/zeroex_celo_api_fills_deduped_sample.csv:1]: Do you have any idea how to fix this issue?

Comment (RantumBits) [reply seeds/zeroex/celo/zeroex_celo_api_fills_deduped_sample.csv:1]: No, I have not been able to identify the problem yet

Comment (RantumBits) [reply seeds/zeroex/celo/zeroex_celo_api_fills_deduped_sample.csv:1]: This is resolved. Had the wrong proxy contract address

**Review by Hosuke — APPROVED** [review]
LGTM.✅
Thank you @RantumBits 

[Sample zeroex_celo_api_fills](https://dune.com/queries/2989206)
[Sample zeroex_celo_api_fills_deduped](https://dune.com/queries/2989202)

**Review by jeff-dude — APPROVED** [review]

**Pull request merged**